modules/lang-painless/src/main/java/org/elasticsearch/painless/node/ERegex.java

```diff
@@ -59,19 +56,14 @@ public ERegex(Location location, String pattern, String flagsString) {
         this.flags = flags;
     }
 
-    @Override
-    void storeSettings(CompilerSettings settings) {
-        this.settings = settings;
-    }
-
     @Override
     void extractVariables(Set<String> variables) {
         // Do nothing.
     }
 
     @Override
     void analyze(ScriptRoot scriptRoot, Locals locals) {
-        if (false == settings.areRegexesEnabled()) {
```

Comment:
Can you change this to `scriptRoot.getCompilerSettings().areRegexesEnabled() == false`

Reply:
Since this change is nearly mechanical, I would prefer to make this change in a follow up PR.